README.md

```diff
@@ -6,14 +6,122 @@ A validator for CSV files adhering to the [Data Models](https://github.com/chop-
+
+```
+$ data-models-validator -model pedsnet -version 2.0.0 foo.csv:person
+```
+
+Run the following to see the full usage:
+
+```
+$ data-models-validator -help
+```
+
+## Functionality
+
+The validator checks the following:
+
+- header matches fields of specified table
+- each row of data has the correct number of fields
+- data is encoded in UTF8
+- quotes within data values are escaped
+- date and datetime data is valid and properly formatted
+- integer and number (float) data is valid and fits in 32-bit types
+- required data is not left null
+- string data does not exceed defined max lengths
+
+The validator does **not** check:
+
+- foreign key referential integrity
+- data model conventions such as correct concept usage
+- uniqueness of data across rows
+
+## Known Bugs
+
+If the validator is run several times in quick succession, an error from the underlying data models service is thrown:
+
+```
+$ data-models-validator -model i2b2_pedsnet -version 2.0.0 obs-fact.csv:observation_fact
+error decoding model revisions: invalid character '<' looking for beginning of value
+```
+
+No harm is done to any data and by waiting several minutes before re-running, the problem can be circumvented. The development team is working on a fix ASAP and you can track our progress [here](https://github.com/chop-dbhi/data-models-validator/issues/9)
+
+## Future Directions
+
+Soon, we hope to add higher level validation checks, especially of referential integrity.
+
+## Output Examples
+
+If the filename does not match a table in the model, a list of known tables in the model is printed. The solution is to specify the table by adding `:table_name` to the end of your file name:
+
+```
+$ data-models-validator -model pedsnet -version 2.0.0 PEDSNET_DRUG_EXPOSURE.csv
+Validating against model 'pedsnet/2.0.0'
+* Unknown table 'PEDSNET_DRUG_EXPOSURE'.
+Choices are: care_site, concept, concept_ancestor, concept_class, concept_relationship, concept_synonym, condition_occurrence, death, domain, drug_exposure, drug_strength, fact_relationship, location, measurement, observation, observation_period, person, procedure_occurrence, provider, relationship, source_to_concept_map, visit_occurrence, visit_payer, vocabulary
+
+$ data-models-validator -model pedsnet -version 2.0.0 PEDSNET_DRUG_EXPOSURE.csv:drug_exposure
+```
+
+If the header does not match the expected set of fields, the expected and actual number of fields as well as any unknown and/or missing fields found in the header are printed:
+
+```
+$ data-models-validator -model pedsnet -version 2.0.0 original_visit_occurrence.csv:visit_occurrence
+Validating against model 'pedsnet/2.0.0'
+* Evaluating 'visit_occurrence' table in 'original_visit_occurrence.csv'...
+* Problem reading CSV header: line 0: [code: 201] Header does not contain the correct set of fields: (expectedLength:12, actualLength:13, unknownFields:[visit_occurrence_source_id], missingFields:[])
```

Comment:
Users don't know *a priori* that they will be provided that information, so not seeing that information translates that nothing else is wrong. If you really want to keep it, ensure that in other error contexts, empty values are not be added.